src/js/components/modals/ServiceFormModal.js

```diff
@@ -269,6 +269,35 @@ class ServiceFormModal extends mixin(StoreMixin) {
   shouldDisableForm(service) {
     let containerSettings = service.getContainerSettings();
 
+    let portDefinitions = service.getPortDefinitions();
+    let name = new RegExp(`${service.getId()}`, 'g');
+
+    if (portDefinitions) {
+      if (!portDefinitions.some(function (port) {
```

Comment:
Maybe we could pull this `some` statement out of the `if` and assign a variable to it. It's a little harder to follow the logic like this.